### ISSUE TITLE
Templates: location information for data-binding is incorrect

### ISSUE THREAD

**Issue by jerone** [angular-eslint/angular-eslint#384]
As part of #292 the template location was fixed for interpolation.

However, the location for **data-binding** is still off.  
Let me explain with the following HTML template:

```html
<div [class]="
    value
">
</div>
```

As seen in the screenshot below, the location of the warning is:

```json
{
    "start": { 
        "line": 2, 
        "column": 9 
    }, 
    "end": { 
        "line": 3, 
        "column": 3 
    }
}
```

```html
<div [class]="
    value
         ~
">
~~
</div>
```

![image](https://user-images.githubusercontent.com/55841/111079390-2016fd00-84fa-11eb-9f7f-617fc44fb74d.png)

While, I expect the error to be under value:

```json
{
    "start": { 
        "line": 2, 
        "column": 4 
    }, 
    "end": { 
        "line": 2, 
        "column": 9 
    }
}
```

```html
<div [class]="
    value
    ~~~~~
">
</div>
```

**Comment by jerone** [comment]
With Angular 13, ESLint 8 and Angular ESLint 13, this is still an issue. 

Wondering which of these 3 packages is responsible for the template location...

**Comment by JamesHenry** [comment]
Hi @jerone, this issue seems to be important to you, so please could you take a look into it? Please remember that this project receives no corporate backing or sponsorship whatsoever, we can make it the best it can be by working together to solve problems as they arise. It's always ideal when folks impacted by issues can work on them themselves

**Comment by jerone** [comment]
Hi @JamesHenry. Sorry, I didn't want to push, just gave an update on the current situation after reproducing the issue with the latest dependencies.

The issue came to light after building my own Angular ESLint rule (WIP): [https://github.com/jerone/eslint-plugin-angular-template-consistent-this.](https://github.com/jerone/eslint-plugin-angular-template-consistent-this.) This rule will report and fix incorrect use of `ThisReceiver` (e.g. `this.variable` vs `variable`) in Angular templates. Even before releasing it officially, I got an [report](https://github.com/jerone/eslint-plugin-angular-template-consistent-this/issues/1)  from someone highlighting the issue that blocked me from releasing it officially.

This issue is the same issue as in #538, but not coupled to that rule only (maybe consider merging both issues). Probably all rules are affected. I'm surprised that it isn't reported more often.

As mentioned in #292, an similar issue got fixed for Interpolation. The [fix](https://github.com/angular-eslint/angular-eslint/commit/737fd04946a9533698c04665c771d944ffbe430c#diff-e811fff81072d77ff68dbb47ca39a5dd0ef99e59358ffc47194fceefb17ecbf8) for that issue appeared to be to enable config options `preserveLineEndings` and `preserveWhitespaces` to the parser. Sadly that wasn't the fix for data-binding.

After some investigating, this issue seems nothing to do with ESLint, Angular ESLint or TypeScript ESLint, but with Angular.  
I saw that Angular ESLint is [parsing](https://github.com/angular-eslint/angular-eslint/blob/main/packages/template-parser/src/index.ts#L234) the Angular templates with Angular [`HtmlParser`](https://github.com/angular/angular/blob/23e7f7f2f35111040411d6d93870ea987d31ce19/packages/compiler/src/render3/view/template.ts#L2068).

How would you tackle this issue?

**Comment by jerone** [comment]
I might have found the reason for the incorrect reporting of LOC positions. It looks like **any whitespaces or line-breaks before the expression** are ignored inside an data-binding.

I've implemented a work-around inside my own rule: [jerone/eslint-plugin-angular-template-consistent-this@fb7c29c#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R88-R118](https://github.com/jerone/eslint-plugin-angular-template-consistent-this/commit/fb7c29c73c3f952fece51a43ef34bcf577ab3184#diff-da10bcc7d8dff796f50c9358a7268eb5abd2c50f460aecc4b2be1c691a966224R88-R118)

**Comment by JamesHenry** [comment]
@jerone Sorry this took so long to finally resolve, I tested my solution locally with your original snippet I was able to make the `no-restricted-syntax` rule perfectly target the word `value` and have the squiggly show up in my IDE accordingly
